### PR TITLE
Adding an option to toggle the consumption of suffix characters

### LINKED ISSUE #468: Replaces text after the cursor when it equals the suggestion
Weird behavior, I think a gif explains it best:

autocomplete-plus-bug

### LINKED ISSUE #559: completions wipe out subsequent word if it's the same
Put the cursor before any word.  Start typing the word following the cursor.  Wait for auto-complete popup and accept that word.  The following word gets wiped out.

clip-text

(`|` is a cursor)
```
|resolve
reso|resolve   # Now select "resolve" from the drop down
resolve|

# expected:
resolve|resolve
```

Atom 1.0.5, OSX 10.10.4

### PR DESCRIPTION
**Pull request by zyklus**
Adds the option:

image

Which, when disabled, ignores any suffix text.

Closes #468
Closes #559

### CODE REVIEW

File: spec/autocomplete-manager-integration-spec.coffee [hunk 1]
```diff
@@ -1179,6 +1180,20 @@ describe 'Autocomplete Manager', ->
...
+          editor.setCursorBufferPosition([0, 2])
+          triggerAutocompletion(editor, false, 'e')
+
+          runs ->
+            suggestionListView = editorView.querySelector('.autocomplete-plus autocomplete-suggestion-list')
+            atom.commands.dispatch(suggestionListView, 'autocomplete-plus:confirm')
+
+            expect(editor.getText()).toBe 'oneomgtwotwothree'
+
+
```
Comment (benogle) [inline spec/autocomplete-manager-integration-spec.coffee]: :fire: empty line

**Comment by benogle** [comment]
Looks like this no longer merges cleanly. Can you update it?

File: lib/main.coffee [hunk 2]
```diff
@@ -106,6 +106,12 @@ module.exports =
       items:
         type: 'string'
       order: 16
+    consumeSuffixWords:
+      title: 'Consume suggestion text following the cursor'
+      description: 'Completing a suggestion will delete all identical text after the cursor'
```
Comment (benogle) [inline lib/main.coffee]: This sounds pretty scary. How about

> Completing a suggestion consumes text following the cursor matching the suffix of the chosen suggestion.

File: lib/main.coffee [hunk 3]
```diff
@@ -106,6 +106,12 @@ module.exports =
       items:
         type: 'string'
       order: 16
+    consumeSuffixWords:
```
Comment (benogle) [inline lib/main.coffee]: Maybe just `consumeSuffix`

File: lib/main.coffee (hunk 2 above)
Comment (zyklus) [inline lib/main.coffee]: It is scary!  And evil!  It wipes out text!  :smile:

**Comment by zyklus** [comment]
updated

**Pull request merged**